src/Stripe.net/Infrastructure/RuntimeInformation.cs

```diff
@@ -1,53 +1,264 @@
+        /// <summary>Returns a string that indicates the name of the .NET installation on which an app is running.</summary>
+        /// <returns>A string that indicates the name of the .NET installation on which an app is running.</returns>
+        public static string GetRuntimeVersion()
+        {
+            if (IsMono)
+            {
+                return GetMonoVersion();
+            }
+            else if (IsFullFramework)
+            {
+                return GetFullFrameworkVersion();
+            }
+            else if (IsNetCore)
+            {
+                return GetNetCoreVersion();
+            }
+            else if (IsCoreRT)
+            {
+                return FrameworkDescription.Replace("Core ", "CoreRT ");
+            }
+
+            return Unknown;
+        }
+
+        internal static string GetMonoVersion()
+        {
+            var monoRuntimeType = Type.GetType("Mono.Runtime");
+            var monoDisplayName = monoRuntimeType?.GetMethod("GetDisplayName", BindingFlags.NonPublic | BindingFlags.Static);
+            if (monoDisplayName != null)
+            {
+                string version = monoDisplayName.Invoke(null, null)?.ToString();
+                if (version != null)
+                {
+                    int bracket1 = version.IndexOf('('), bracket2 = version.IndexOf(')');
+                    if (bracket1 != -1 && bracket2 != -1)
+                    {
+                        string comment = version.Substring(bracket1 + 1, bracket2 - bracket1 - 1);
+                        var commentParts = comment.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+                        if (commentParts.Length > 2)
+                        {
+                            version = version.Substring(0, bracket1) + "(" + commentParts[0] + " " + commentParts[1] + ")";
+                        }
+                    }
+                }
+
+                return "Mono " + version;
+            }
+
+            return Unknown;
+        }
+
+        internal static string GetFullFrameworkVersion()
         {
-#if NET45
-            return Environment.OSVersion.ToString();
-#else
-            return System.Runtime.InteropServices.RuntimeInformation.OSDescription;
-#endif
+            var fullName = System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription; // sth like .NET Framework 4.7.3324.0
+            var servicingVersion = new string(fullName.SkipWhile(c => !char.IsDigit(c)).ToArray());
+            var releaseVersion = MapToReleaseVersion(servicingVersion);
+
+            return $".NET Framework {releaseVersion}";
         }
 
-#if NET45
-        public static string GetMonoVersion()
+        internal static string MapToReleaseVersion(string servicingVersion)
         {
-            Type monoRuntimeType = typeof(object).Assembly.GetType("Mono.Runtime");
+            // the following code assumes that .NET 4.5 is the oldest supported version
```

Comment:
I don't fully grasp why we do this instead of sending `servicingVersion` or sanitizing it ourselves?

Reply:
The "servicing version" includes patch/build numbers that are not particularly relevant, so this method (adapted from [here](https://github.com/dotnet/BenchmarkDotNet/blob/4740478f46f3c98d97d4a5b470260ebd7efa571b/src/BenchmarkDotNet/Helpers/FrameworkVersionHelper.cs#L38-L53)) sanitizes the servicing version to an actual release version (https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/versions-and-dependencies#version-information).